libdnf/sack/query.cpp

```diff
@@ -1006,11 +1016,17 @@ Query::Impl::filterNevraStrict(int cmpType, const char **matches)
     const unsigned nmatches = g_strv_length((gchar**)matches);
     compareSet.reserve(nmatches);
     NevraID nevraId;
+    
+    int createNewEVR = 0;
+    if (cmpType & HY_LT || cmpType & HY_GT) {
+        createNewEVR = 1;
+    }
+    
     for (unsigned int i = 0; i < nmatches; ++i) {
         const char * nevraPattern = matches[i];
         if (!nevraPattern)
             throw std::runtime_error("Query can not accept NULL for STR match");
-        if (nevraId.parse(pool, nevraPattern)) {
+        if (nevraId.parse(pool, nevraPattern, createNewEVR)) {
```

Comment:
If `createNewEVR` is `true`, a new string may be created in the pool. Who will remove it from the pool? I think this is not a problem now. Only note.